### PR TITLE
DDF-2110 The XACML PDP should attempt to determine the value types

### PR DESCRIPTION
**Pull request by bcwaters**
#### What does this PR do?
Updates the XACML PDP to attempt to determine the value types for resource and subject attributes

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@roelens8 @ryeats 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@shaundmorris 
@stustison

#### How should this be tested?
Full build on ddf.

#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: platform/security/pdp/security-pdp-authzrealm/pom.xml [hunk 1]
```diff
@@ -83,6 +83,11 @@
             <groupId>ddf.platform.util</groupId>
             <artifactId>platform-util</artifactId>
         </dependency>
+        <dependency>
+            <groupId>commons-validator</groupId>
```
Comment (stustison) [inline platform/security/pdp/security-pdp-authzrealm/pom.xml]: we probably need to be embedding this

Comment (bcwaters) [inline platform/security/pdp/security-pdp-authzrealm/pom.xml]: updated

**Comment by tbatie** [comment]
:+1: Hero'd successfully

File: platform/security/pdp/security-pdp-authzrealm/pom.xml [hunk 2]
```diff
@@ -83,6 +83,16 @@
...
             <artifactId>platform-util</artifactId>
         </dependency>
+        <dependency>
+            <groupId>oro</groupId>
+            <artifactId>oro</artifactId>
+            <version>2.0.8</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.servicemix.bundles</groupId>
+            <artifactId>org.apache.servicemix.bundles.commons-validator</artifactId>
```
Comment (stustison) [inline platform/security/pdp/security-pdp-authzrealm/pom.xml]: Shouldn't have a dependency on the servicemix bundle. This should be a dependency on the commons-validator jar instead.

File: platform/security/security-services-app/src/main/resources/features.xml [hunk 3]
```diff
@@ -61,6 +61,8 @@
         <feature prerequisite="true">security-expansion</feature>
         <feature prerequisite="true">security-expansion-user-attributes</feature>
         <feature prerequisite="true">security-expansion-metacard-attributes</feature>
+        <bundle dependency="true">mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.commons-beanutils/1.8.3_2</bundle>
```
Comment (stustison) [inline platform/security/security-services-app/src/main/resources/features.xml]: Is it absolutely necessary to include this bundle? There is a CVE against commons-beanutils and I'd rather not include it here.

Comment (bcwaters) [inline platform/security/security-services-app/src/main/resources/features.xml]: fixed

File: platform/security/pdp/security-pdp-authzrealm/pom.xml (hunk 2 above)
Comment (bcwaters) [inline platform/security/pdp/security-pdp-authzrealm/pom.xml]: fixed

File: platform/security/pdp/security-pdp-authzrealm/src/main/java/ddf/security/pdp/realm/xacml/XACMLConstants.java [hunk 4]
```diff
@@ -22,6 +22,10 @@
 
     public static final String BOOLEAN_DATA_TYPE = "http://www.w3.org/2001/XMLSchema#boolean";
 
+    public static final String INTEGER_DATA_TYPE = "http://www.w3.org/2001/XMLSchema#integer";
```
Comment (shaundmorris) [inline platform/security/pdp/security-pdp-authzrealm/src/main/java/ddf/security/pdp/realm/xacml/XACMLConstants.java]: This was already here, but we should removing XACMLConstants (okay with it being another ticket) and moving the constants to XacmlPdp. The constants are only being used by XacmlPdp anyway.

**Pull request merged**